### PR TITLE
Create Decision Review Updated Claim Review Builder

### PR DESCRIPTION
**Pull request by gcd253**
Resolves [APPEALS-54019](https://jira.devops.va.gov/browse/APPEALS-54019)

# Description
- Adds new class `DecisionReviewUpdated::ClaimReview` which allows reader and writer access to its attributes: `same_office`, `informal_conference`, and `legacy_opt_in_approved`

- Adds new class `Builders::DecisionReviewUpdated::ClaimReviewBuilder` which assigns the following attributes: `same_office`, `informal_conference`, and `legacy_opt_in_approved` to an instance of the `DecisionReviewUpdated::ClaimReview` class

- Adds `decision_review_updated_claim_review` factory with traits for each of its attributes: `same_office`, `informal_conference`, and `legacy_opt_in_approved`

- Reworks DecisionReviewUpdated AVRO to define `Decision` and to include several other attributes

## Acceptance Criteria
- [x] Code compiles correctly
- [x] A new class named `Builders::DecisionReviewUpdated::ClaimReviewBuilder` exists within the following directory: `app/models/builders/decision_review_updated`
- [x] A new class named `DecisionReviewUpdated::ClaimReview`
 exists within the following directory: `app/models/virtual/decision_review_updated`

## Testing Plan

1. [APPEALS-54476](https://jira.devops.va.gov/browse/APPEALS-54476)
2. [APPEALS-55519](https://jira.devops.va.gov/browse/APPEALS-55519)
3. [Jira Xray](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-55519&testIssueKey=APPEALS-54476)

# Backend

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/models/builders/decision_review_updated/claim_review_builder.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  # Combines decision review issues of all types into single array
+  def combine_decision_review_issues
+    all_decision_review_issues = []
+    decision_review_issues.each { |issues| all_decision_review_issues += issues }
+    all_decision_review_issues.flatten
+  end
+
+  def decision_review_issues
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_updated/claim_review_builder.rb:49]: rename this to all_decision_review_updated_issues

File: app/models/builders/decision_review_updated/claim_review_builder.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    [
+      decision_review_updated.decision_review_issues_created,
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_updated/claim_review_builder.rb:51]: make a private method for each of the arrays within this 2 dimiensional array

File: spec/models/virtual/decision_review_updated/claim_review_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+describe DecisionReviewUpdated::ClaimReview do
```
Comment (Aaron-Willis) [inline spec/models/virtual/decision_review_updated/claim_review_spec.rb:3]: Test instantion of the object

File: app/models/builders/decision_review_updated/claim_review_builder.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# This class is used to build out a Claim Review object from an instance of DecisionReviewUpdated
+class Builders::DecisionReviewUpdated::ClaimReviewBuilder
+  attr_reader :claim_review, :decision_review_updated
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_updated/claim_review_builder.rb:5]: do you need to have claim_review as an attr_reader?

Comment (gcd253) [reply app/models/builders/decision_review_updated/claim_review_builder.rb:5]: claim_review as an attr_reader here allows us to return `builder.claim_review` at the end of the build method on line 10

**Review by Aaron-Willis — APPROVED** [review]

**Pull request merged**